### PR TITLE
Can define upgrade

### PR DESCRIPTION
**Pull request by obaidott**
- [x]   Fix/update all test cases
- [x]  Add support for can-define-stream
- [x]  Update to 3.0

### CODE REVIEW

File: public/models/stat.js [hunk 1]
```diff
@@ -24,15 +24,15 @@
...
-	set = require("can-set"),
-	can = require("can/util/");
+	set = require("can-set");
+var DefineMap = require('can-define/map/map');
+var DefineList = require('can-define/list/list');
+var Player = require("bitballs/models/player");
 
-require("can/map/define/");
-require("can/list/");
+require("can-map-define");
```
Comment (justinbmeyer) [inline public/models/stat.js]: remove

File: public/models/team.js [hunk 2]
```diff
@@ -8,124 +8,138 @@ var superMap = require('can-connect/can/super-map/');
 var tag = require('can-connect/can/tag/');
 var set = require("can-set");
 var Player = require("./player");
-var can = require("can/util/");
-require("can/map/define/");
-require("can/list/");
+var DefineMap = require("can-define/map/map");
+var DefineList = require("can-define/list/list");
+require("can-map-define");
```
Comment (justinbmeyer) [inline public/models/team.js]: remove

File: public/models/team.js [hunk 3]
```diff
@@ -8,124 +8,138 @@ var superMap = require('can-connect/can/super-map/');
...
-				var players = [],
-					self = this;
+	/**
+	 * @property {Number} bitballs/models/team.properties.tournamentId tournamentId
+	 * @parent bitballs/models/team.properties
+	 *
+	 * The `id` of [bitballs/models/tournament] that the team will be
+	 * associated with.
+	 **/
+	tournamentId: {
```
Comment (justinbmeyer) [inline public/models/team.js]: `tournamentId: "number"` ... please fix others too.

File: public/models/team.js [hunk 4]
```diff
@@ -8,124 +8,138 @@ var superMap = require('can-connect/can/super-map/');
...
+	/**
+	 * @property {bitballs/models/player.static.List} bitballs/models/team.properties.players players
+	 * @parent bitballs/models/team.properties
+	 *
+	 * A list made up of the [bitballs/models/player] models referenced
+	 * by properties [bitballs/models/team.properties.player1],
+	 * [bitballs/models/team.properties.player2], [bitballs/models/team.properties.player3],
+	 * and [bitballs/models/team.properties.player4].
+	 **/
+	players: {
```
Comment (justinbmeyer) [inline public/models/team.js]: getter shorthand

File: public/models/team.js [hunk 5]
```diff
@@ -8,124 +8,138 @@ var superMap = require('can-connect/can/super-map/');
...
-		idMap: {
-			type: "*",
-			get: function(){
-				var map = {};
+	/**
+	 * @property {Object}
+	 *
+	 * A map of team ids to [bitballs/models/team] models.
+	 **/
+	idMap: {
```
Comment (justinbmeyer) [inline public/models/team.js]: getter shorthand

File: public/models/team.js [hunk 6]
```diff
@@ -8,124 +8,138 @@ var superMap = require('can-connect/can/super-map/');
...
+		type: 'number',
+	}
 });
 /**
- * @constructor {can.List} bitballs/models/team.static.List List
+ * @constructor {can-list} bitballs/models/team.static.List List
  * @parent bitballs/models/team.static
  */
-Team.List = can.List.extend({Map: Team},
+Team.List = DefineList.extend({Map: Team},
```
Comment (justinbmeyer) [inline public/models/team.js]: Use `#`

File: public/models/tournament.js [hunk 7]
```diff
@@ -8,54 +8,59 @@ var superMap = require('can-connect/can/super-map/');
 var tag = require('can-connect/can/tag/');
 var set = require("can-set");
 var moment = require('moment');
-var can = require("can/util/");
-require("can/map/define/");
-require("can/list/");
+var DefineMap = require("can-define/map/map");
+var DefineList = require("can-define/list/list");
+require("can-map-define");
```
Comment (justinbmeyer) [inline public/models/tournament.js]: remove

File: public/models/tournament.js [hunk 8]
```diff
@@ -8,54 +8,59 @@ var superMap = require('can-connect/can/super-map/');
...
+var Tournament = DefineMap.extend('Tournament', {
+	id: 'number',
+	/**
+	 * @property {Date} bitballs/models/tournament.properties.jsDate jsDate
+	 * @parent bitballs/models/tournament.properties
+	 *
+	 * The [bitballs/models/tournament.properties.date] converted to a
+	 * JavaScript Date object.
+	 **/
+	jsDate: {
```
Comment (justinbmeyer) [inline public/models/tournament.js]: getter shorthand, please update others

File: public/models/tournament.js [hunk 9]
```diff
@@ -8,54 +8,59 @@ var superMap = require('can-connect/can/super-map/');
...
 	}
 });
 
 /**
- * @constructor {can.List} bitballs/models/tournament.static.List List
+ * @constructor {can-list} bitballs/models/tournament.static.List List
  * @parent bitballs/models/tournament.static
  */
-Tournament.List = can.List.extend({Map: Tournament},{});
+Tournament.List = DefineList.extend({Map: Tournament},{});
```
Comment (justinbmeyer) [inline public/models/tournament.js]: Use `#`

File: public/models/user.js [hunk 10]
```diff
@@ -25,17 +25,39 @@
 var superMap = require('can-connect/can/super-map/');
 var tag = require('can-connect/can/tag/');
 var set = require("can-set");
-var can = require("can/util/");
-require("can/map/define/");
-require("can/list/");
+var DefineMap = require('can-define/map/map');
+var DefineList = require("can-define/list/list");
+require("can-map-define");
```
Comment (justinbmeyer) [inline public/models/user.js]: remove

File: public/models/user.js [hunk 11]
```diff
@@ -25,17 +25,39 @@
...
 
-var User = can.Map.extend({});
+var User = DefineMap.extend('User', {
+	email: {
```
Comment (justinbmeyer) [inline public/models/user.js]: `email: "string"`.  Please update others.

File: public/models/user.js [hunk 12]
```diff
@@ -25,17 +25,39 @@
...
+	}
+});
 
 /**
- * @constructor {can.List} bitballs/models/user.static.List List
+ * @constructor {can-list} bitballs/models/user.static.List List
  * @parent bitballs/models/user.static
  */
-User.List = can.List.extend({Map: User},{});
+User.List = DefineList.extend({Map: User},{});
```
Comment (justinbmeyer) [inline public/models/user.js]: Use `#`

File: public/package.json [hunk 13]
```diff
@@ -15,34 +15,55 @@
...
+    "can": "^3.0.0-pre.18",
+    "can-component": "^3.0.2",
+    "can-connect": "^1.0.6",
+    "can-construct": "^3.0.1",
+    "can-define": "^1.0.3",
+    "can-define-backup": "0.0.1",
+    "can-define-stream": "0.0.2",
+    "can-list": "^3.0.0",
+    "can-map": "^3.0.2",
+    "can-map-define": "^3.0.0",
```
Comment (justinbmeyer) [inline public/package.json]: We should be able to remove `can-list`, `can-map`, `can-map-define`.

File: public/package.json [hunk 14]
```diff
@@ -15,34 +15,55 @@
...
+    "done-ssr-middleware": "^0.4.0-pre.0",
     "funcunit": "^3.1.0-pre.1",
     "generator-donejs": "0.6.4",
     "jquery": "^2.1.4",
     "moment": "^2.10.6",
-    "steal": "^0.15.0",
+    "steal": "^0.16.41",
+    "steal-platform": "0.0.4",
     "steal-qunit": "^0.1.1",
+    "steal-stache": "^3.0.0-pre.3",
```
Comment (justinbmeyer) [inline public/package.json]: there should be a 3.0.0 version of this now.

File: public/package.json [hunk 15]
```diff
@@ -15,34 +15,55 @@
...
     "steal-tools": "^0.15.0",
```
Comment (justinbmeyer) [inline public/package.json]: should be a 0.16.0 veriosn.

File: public/package.json [hunk 16]
```diff
@@ -15,34 +15,55 @@
...
-    "steal-platform": "0.0.4",
-    "done-ssr-middleware": "^0.1.0",
     "yeoman-environment": "^1.2.7"
   },
   "devDependencies": {
-    "donejs-cli": "^0.6.0",
-    "can-fixture": "^0.1.0",
+    "donejs-cli": "^0.10.0-pre.0",
+    "can-fixture": "^0.4.0-pre.2",
```
Comment (justinbmeyer) [inline public/package.json]: there should be a 1.0.0 version of this.

File: public/package.json [hunk 17]
```diff
@@ -15,34 +15,55 @@
...
+    "can-view-model": "^3.0.0",
+    "can-view-target": "^3.0.3",
+    "done-autorender": "^0.9.0-pre.0",
+    "done-component": "^0.6.0-pre.1",
+    "done-css": "~2.1.0-pre.0",
+    "done-serve": "^0.3.0-pre.0",
+    "done-ssr-middleware": "^0.4.0-pre.0",
     "funcunit": "^3.1.0-pre.1",
     "generator-donejs": "0.6.4",
     "jquery": "^2.1.4",
```
Comment (justinbmeyer) [inline public/package.json]: ideally, we can remove jQuery

File: public/package.json [hunk 18]
```diff
@@ -15,34 +15,55 @@
...
     "testee": "^0.2.4"
   },
   "system": {
     "configDependencies": [
       "node_modules/can-zone/register",
       "live-reload"
     ],
+    "map": {
+      "can/util/jquery/jquery": "jquery"
+    },
```
Comment (justinbmeyer) [inline public/package.json]: we should be able to remove this.

File: public/util/numberToString.js [hunk 19]
```diff
@@ -1,11 +1,10 @@
...
-can.stache.registerHelper("numberToString", function(newVal, source){
-	if(newVal instanceof can.expression.SetIdentifier) {
-		source(newVal.value === "" ? null : +newVal.value);
-	} else {
-		source = newVal;
-		return source() + "";
-	}
-});
\ No newline at end of file
+stache.registerConverter("numberToString",{
```
Comment (justinbmeyer) [inline public/util/numberToString.js]: I think there is a converter like this already.  We should try to use that.

File: public/util/numberToString_test.js [hunk 20]
```diff
@@ -1,31 +1,38 @@
...
 
 QUnit.module('util/numberToString');
 
 QUnit.test("numberToString helper", function(assert){
 
-	var template = can.view.stache('<input {($value)}="numberToString(~age)">');
+	var template = stache('<input {($value)}="numberToString(~age)">');
 
-	var map = new can.Map({age: 25});
+	var map = new CanMap({age: 25});
```
Comment (justinbmeyer) [inline public/util/numberToString_test.js]: lets get rid of CanMap.

File: public/util/numberToString.js (hunk 19 above)
Comment (obaidott) [inline public/util/numberToString.js]: Do you know where can I find documentation for the converter? Can't seem to find anything on the http://v3.canjs.com/doc site.

Comment (justinbmeyer) [inline public/util/numberToString.js]: http://v3.canjs.com/doc/can-stache-converters.html

Comment (obaidott) [inline public/util/numberToString.js]: None of those support something similar to `numberToString`, there is `string-to-any` but that's not what we want here.

File: public/models/player.js [hunk 21]
```diff
@@ -1,151 +1,129 @@
...
 	}
 });
 
 /**
- * @constructor {can.List} bitballs/models/player.static.List List
+ * @constructor {can-list} bitballs/models/player.static.List List
  * @parent bitballs/models/player.static
  */
-Player.List = can.List.extend({Map: Player},
+Player.List = DefineList.extend('PlayerList', {"#": Player},
```
Comment (justinbmeyer) [inline public/models/player.js]: this is supposed to be in the prototype, not the static.  This seems to be a few places.

File: public/components/player/edit/edit.js [hunk 22]
```diff
@@ -28,42 +28,45 @@
...
+	},
+	/**
+	* @property {bitballs/models/player} bitballs/components/player/edit.player player
+	* @parent bitballs/components/player/edit.properties
+	*
+	* The model that will be bound to the form.
+	**/
+	player: {
+		Type: Player,
+		value: Player
```
Comment (justinbmeyer) [inline public/components/player/edit/edit.js]: this getting changed to lower case `value` broke the edit player form.

**Pull request closed without merging**